opt/jdbc.sh

```diff
@@ -4,61 +4,60 @@ set_jdbc_url() {
+  if [ "$db_protocol" = "postgres" ]; then
+    local jdbc_protocol="jdbc:postgresql"
 
-        if [ "${CI:-}" != "true" ]; then
-          local db_default_args="&sslmode=require"
-        else
-          local db_default_args=""
-        fi
-      elif [ "$db_protocol" = "mysql" ]; then
-        local jdbc_protocol="jdbc:mysql"
-      fi
+    if [ "${CI:-}" != "true" ]; then
+      local db_default_args="&sslmode=require"
+    else
+      local db_default_args=""
+    fi
+  elif [ "$db_protocol" = "mysql" ]; then
+    local jdbc_protocol="jdbc:mysql"
+  fi
 
-      if [ -n "$jdbc_protocol" ]; then
-        local db_user
-        db_user=$(expr "$db_url" : "${db_protocol}://\(.\+\):\(.\+\)@")
+  if [ -n "$jdbc_protocol" ]; then
+    local db_user
+    db_user=$(expr "$db_url" : "${db_protocol}://\(.\+\):\(.\+\)@")
 
-        local db_prefix="${db_protocol}://${db_user}:"
+    local db_prefix="${db_protocol}://${db_user}:"
 
-        local db_pass
-        db_pass=$(expr "$db_url" : "${db_prefix}\(.\+\)@")
-        db_prefix="${db_prefix}${db_pass}@"
+    local db_pass
+    db_pass=$(expr "$db_url" : "${db_prefix}\(.\+\)@")
+    db_prefix="${db_prefix}${db_pass}@"
 
-        local db_host_port
-        db_host_port=$(expr "$db_url" : "${db_prefix}\(.\+\)/")
-        db_prefix="${db_prefix}${db_host_port}/"
+    local db_host_port
+    db_host_port=$(expr "$db_url" : "${db_prefix}\(.\+\)/")
+    db_prefix="${db_prefix}${db_host_port}/"
 
-        local db_suffix
-        db_suffix=$(expr "$db_url" : "${db_prefix}\(.\+\)")
+    local db_suffix
+    db_suffix=$(expr "$db_url" : "${db_prefix}\(.\+\)")
 
-        if echo "$db_suffix" | grep -qi "?"; then
-          local db_args="&user=${db_user}&password=${db_pass}"
-        else
-          local db_args="?user=${db_user}&password=${db_pass}"
-        fi
+    if echo "$db_suffix" | grep -qi "?"; then
+      local db_args="&user=${db_user}&password=${db_pass}"
+    else
+      local db_args="?user=${db_user}&password=${db_pass}"
+    fi
 
-        if [ -n "$db_host_port" ] &&
-             [ -n "$db_suffix" ] &&
-             [ -n "$db_user" ] &&
-             [ -n "$db_pass" ]; then
-          eval "export ${env_prefix}_URL=\"${jdbc_protocol}://${db_host_port}/${db_suffix}${db_args}${db_default_args}\""
-          eval "export ${env_prefix}_USERNAME=\"${db_user}\""
-          eval "export ${env_prefix}_PASSWORD=\"${db_pass}\""
-        fi
-      fi
+    if [ -n "$db_host_port" ] &&
+      [ -n "$db_suffix" ] &&
+      [ -n "$db_user" ] &&
+      [ -n "$db_pass" ]; then
+      eval "export ${env_prefix}_URL=\"${jdbc_protocol}://${db_host_port}/${db_suffix}${db_args}${db_default_args}\""
+      eval "export ${env_prefix}_USERNAME=\"${db_user}\""
+      eval "export ${env_prefix}_PASSWORD=\"${db_pass}\""
+    fi
   fi
 }
 
 if [ -n "${DATABASE_URL:-}" ]; then
   set_jdbc_url "$DATABASE_URL"
   if [ -n "${DATABASE_CONNECTION_POOL_URL:-}" ]; then
     set_jdbc_url "$DATABASE_CONNECTION_POOL_URL"
```

Comment:
If the goal is to keep `JDBC_DATABASE_URL` and `DATABASE_URL` consistent then this should be removed as it sets `JDBC_DATABASE_URL` to use the pooling postgres port.  Calling `set_jdbc_url` second time without a prefix being passed overrides `JDBC_DATABASE*` environment variables.

Reply:
That's the intention. I want to keep it so it does work the way it's currently documented. We source the connection for `JDBC_DATABASE_URL` from other config vars than `DATABASE_URL` (from `JAWSDB_URL` for example) already.

I agree it's hard to keep track what `JDBC_DATABASE_URL` contains in more complex scenarios (that's why I added more tests), but I don't think this this PR worsens the situation.

Reply:
👍